decidim-core/lib/decidim/core/engine.rb

```diff
@@ -382,9 +382,18 @@ class Engine < ::Rails::Engine
       end
 
       initializer "decidim_core.notifications" do
-        config.to_prepare do
-          Decidim::EventsManager.subscribe(/^decidim\.events\./) do |event_name, data|
-            EventPublisherJob.perform_later(event_name, data)
+        if Rails.version > "6.0" && Rails.autoloaders.zeitwerk_enabled?
```

Comment:
I think this condition will always be truth, as we don't support any older Rails version currently, and zeitwerk it's the only Rails autoloader as far as I know

Reply:
Fixed here: [5431229](https://github.com/decidim/decidim/pull/12485/commits/543122973882e6a268fe6cc8ccee258b442fc23e)

I have removed the "Rails.version > 6.0 " condition as we have a strict requirement for rails to be greater than 6.0. I have left the `Rails.autoloaders.zeitwerk_enabled?` part, as the latest rails documentation, reveals that even 7.1 still support the switch to classic mode. 

https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#autoloading-and-engines